modules/apps/static/portal-lpkg-deployer/portal-lpkg-deployer-impl/src/main/java/com/liferay/portal/lpkg/deployer/internal/LPKGBundleTrackerCustomizer.java

```diff
@@ -527,13 +529,19 @@ private String _buildImportPackageString(Class<?>... classes) {
 	}
 
 	private String _extractFileName(String string) {
-		int endIndex = string.indexOf(CharPool.DASH);
+		Matcher matcher = _pattern.matcher(string);
 
-		int beginIndex = string.lastIndexOf(CharPool.SLASH, endIndex) + 1;
+		if (matcher.matches()) {
+			String name = matcher.group(1);
 
-		String name = string.substring(beginIndex, endIndex);
+			return name.concat(string.substring(string.length() - 4));
```

Comment:
@mtambara this ```string.substring(string.length() - 4)``` looks odd. I think you meant for reattaching the ".jar" postfix. But the regrex did not test for that postfix, so we don't really know what it is. If you meant to preserve the postfix, no matter what it is, you will need to find the last '.' to chop it.